File: Spring-Boot-Ecommerce/src/main/java/com/backend/ecommerce/dto/Purchase.java
package com.backend.ecommerce.dto;

import java.util.HashSet;
import java.util.Set;

import com.backend.ecommerce.entity.Address;
import com.backend.ecommerce.entity.Customer;
import com.backend.ecommerce.entity.OrderItem;
import com.backend.ecommerce.entity.Order;

import lombok.Data;

@Data
public class Purchase {

	private Customer customer;
	private Address shippingAddress;
	private Address billingAddress;
	private Order order;
	private Set<OrderItem> orderItems;
}

File: Angular-FrontEnd/angular-ecommerce/src/app/commons/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { OrderItems } from "./order-item";
import { Order } from "./orders";

export class Purchase {
    customer: Customer;
    shippingAddress: Address;
    billingAddress: Address;
    order: Order;
    orderItems: OrderItems[];
}

